root/static/scripts/common/utility/isDateEmpty.js

```diff
@@ -7,6 +7,17 @@
  * later version: http://www.gnu.org/licenses/gpl-2.0.txt
  */
 
+export function isDateObservableEmpty(
+  date: {
+    day: KnockoutObservable<number | null>,
+    month: KnockoutObservable<number | null>,
+    year: KnockoutObservable<number | null>,
+  },
+): boolean {
+  return (date == null) ||
+    (date.year() == null && date.month() == null && date.day() == null);
```

Comment:
Those are generally strings, aren't they? So I'm not sure that they wouldn't be `''` if empty. I'd double check, but we perform the `tooShortYear` checks on the same observable AFAICT, which expects a string.
```suggestion
    !(nonEmpty(date.year()) || nonEmpty(date.month()) || nonEmpty(date.day());
```

Reply:
Hmm. This seems to work as is, which made me assume that this worked? But I can make the change and see.

Reply:
Checked, printed, etc. - it's null, not empty string.

Reply:
Ok. Thanks for checking. :) Though I do slightly prefer `nonEmpty` in the off chance that there's some weird code path that sets it to an empty string.

Reply:
Ok, can make the change :)